src/integrationtests/java/com/aws/greengrass/integrationtests/metrics/MetricsEmitterTest.java

```diff
@@ -0,0 +1,192 @@
+    private GreengrassServiceClientFactory clientFactory;
+    @Mock
+    private GreengrassV2DataClient client;
+
+    @TempDir
+    Path rootDir;
+
+    //Result captor class created to save and verify the results of scheduled emitting calls
+    private class ResultCaptor<T> implements Answer {
+        @Getter
+        private T result = null;
+
+        @Override
+        public T answer(InvocationOnMock invocationOnMock) throws Throwable {
+            result = (T) invocationOnMock.callRealMethod();
+            return result;
+        }
+    }
+
+    @BeforeEach
+    void beforeEach(ExtensionContext context) throws DeviceConfigurationException {
+        ignoreExceptionOfType(context, SdkClientException.class);
+        ignoreExceptionOfType(context, SpoolerStoreException.class);
+        ignoreExceptionOfType(context, NoSuchFileException.class); // Loading CA keystore
+
+        //Set this property for kernel to scan its own classpath to find plugins
+        System.setProperty("aws.greengrass.scanSelfClasspath", "true");
+        clock = Clock.systemUTC();
+        metricSpy = spy(new ClientDeviceAuthMetrics(clock));
+        domainEvents = new DomainEvents();
+        kernel = new Kernel();
+        kernel.getContext().put(DomainEvents.class, domainEvents);
+        kernel.getContext().put(Clock.class, clock);
+        kernel.getContext().put(ClientDeviceAuthMetrics.class, metricSpy);
+        kernel.getContext().put(GreengrassServiceClientFactory.class, clientFactory);
+        lenient().when(clientFactory.fetchGreengrassV2DataClient()).thenReturn(client);
+    }
+
+    @AfterEach
+    void afterEach() {
+        if (kernel != null) {
+            kernel.shutdown();
+        }
+    }
+
+    private void startNucleusWithConfig(String configFileName) throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
+                getClass().getResource(configFileName).toString());
+        kernel.getContext().addGlobalStateChangeListener((service, was, newState) -> {
+            if (ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME.equals(service.getName())
+                    && service.getState().equals(State.RUNNING)) {
+                authServiceRunning.countDown();
+            }
+        });
+        kernel.launch();
+        assertThat(authServiceRunning.await(TEST_TIME_OUT_SEC, TimeUnit.SECONDS), is(true));
+    }
+
+    private Metric buildMetric(String name) {
+        return Metric.builder()
+                .namespace(NAMESPACE)
+                .name(name)
+                .unit(TelemetryUnit.Count)
+                .aggregation(TelemetryAggregation.Sum)
+                .value(1L)
+                .timestamp(Instant.now(clock).toEpochMilli())
+                .build();
+    }
+
+    @Test
+    void GIVEN_kernelRunningWithMetricsConfig_WHEN_launched_THEN_metricsCorrectlyEmittedAtAggregationInterval()
```

Comment:
this doesn't really seem to be an integration test. You're manually emitting the events, these events aren't being emitting through the normal code path which a customer would execute.

Reply:
My goal with this test was to verify that the emitter schedules correctly and runs as expected, but I can move it to the unit tests if needed

Reply:
Just testing scheduling seems to not provide value. It sounds like you do have a plan for additional testing, let's get that in.

Reply:
I went ahead and added in the tests for SubscribeToCertificateUpdates, VerifyClientDeviceIdentity, and AuthorizeClientDeviceAction. Still working on the GetClientDeviceAuthToken test